frontend/src/features/permits/pages/Amend/types/AmendPermitFormData.ts

```diff
@@ -119,95 +99,53 @@ export const getDefaultFromNullableFormData = (
 ): AmendPermitFormData => {
   const defaultPermitType = getDefaultRequiredVal(DEFAULT_PERMIT_TYPE, permitFormData?.permitType);
 
-  // Default form values when permit is not available (or period of time when loading)
+  // Default form values when form data is not available (or period of time when loading)
   if (!permitFormData) {
     return {
       ...getDefaultValues(defaultPermitType),
-      permitApplicationOrigin: PERMIT_APPLICATION_ORIGINS.ONLINE,
-      permitApprovalSource: PERMIT_APPROVAL_SOURCES.AUTO,
     };
   }
 
   return {
-    ...permitFormData,
-    previousRevision: getDefaultRequiredVal(
-      "",
-      permitFormData.previousRevision,
+    ...getDefaultValues(
+      defaultPermitType,
+      permitFormData,
+      permitFormData.companyId,
     ),
-    comment: getDefaultRequiredVal("", permitFormData.comment),
-    userGuid: getDefaultRequiredVal("", permitFormData.userGuid),
-    documentId: getDefaultRequiredVal("", permitFormData.documentId),
-    permitData: {
-      ...permitFormData.permitData,
-      companyName: getDefaultRequiredVal(
-        "",
-        permitFormData.permitData?.companyName,
-      ),
-      clientNumber: getDefaultRequiredVal(
-        "",
-        permitFormData.permitData?.clientNumber,
-      ),
-    },
   };
 };
 
-export const arePermitsEqual = (
-  permit1?: Nullable<AmendPermitFormData>,
-  permit2?: Nullable<AmendPermitFormData>,
+export const areFormDataEqual = (
```

Comment:
There's potential for using a generic or a recursive function for deep comparisons or perhaps a library rather than do it painstakingly for what seems like every single field.

Reply:
Agreed, luckily I don't think we're actually using this function so I can remove it if you want...

Reply:
+1 for removing it if it is unused. We can always git blame and get back this function in future if needed.